src/nameresolution/builtin.rs

```diff
@@ -179,3 +184,51 @@ fn define_pair(cache: &mut ModuleCache) {
     cache.definition_infos[id.0].typ = Some(constructor_type);
     cache.definition_infos[id.0].definition = Some(constructor);
 }
+
+/// The builtin Maybe type is defined here as:
+///
+/// type Maybe a = 
+///    | Some a
+///    | None
+fn define_maybe(cache: &mut ModuleCache) {
+    let location = Location::builtin();
+
+    let level = LetBindingLevel(0);
+    let a = cache.next_type_variable_id(level);
+
+    let name = "Maybe".to_string();
+    let maybe = cache.push_type_info(name.clone(), vec![], location);
+    assert_eq!(maybe, MAYBE_TYPE);
+
+    cache.type_infos[maybe.0].body = TypeInfoBody::Struct(vec![
+        Field { name: "value".into(), field_type: Type::TypeVariable(a), location },
```

Comment:
Maybe should be a union type, not a struct type. Since its value can be either Some a or None